pkg/kv/kvpb/api.proto

```diff
@@ -828,6 +828,15 @@ enum ChecksumMode {
 message CheckConsistencyRequest {
   RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
   ChecksumMode mode = 3;
+
+  // If enabled, take a storage checkpoint of inconsistent ranges (but do not
+  // fatal nodes). Used with CHECK_FULL for crdb_internal.check_consistency(),
+  // this is done unconditionally with CHECK_VIA_QUEUE.
+  //
+  // TODO(erikgrinaker): the checkpointing should be moved out of the method,
+  // see TODO on CHECK_VIA_QUEUE and Replica.CheckConsistency.
+  bool checkpoint = 6;
```

Comment:
It will break in mixed version clusters. I'd like to backport this to 23.2 for additional test coverage, if possible.

The TODOs in this area also point towards moving this kind of logic out of `ChecksumMode`, so I think it makes sense for the caller to specify different combinations of options as needed.